NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1940,3 +1998,79 @@ def get_ray_path(self, iS):
         return self.__r2d.get_path_reflections(self.__x1, self.__x2, self.__results[iS]['C0'], 10000,
                                    reflection=self.__results[iS]['reflection'],
                                    reflection_case=self.__results[iS]['reflection_case'])
+
+    def create_output_data_structure(self, dictionary, n_showers, n_antennas):
+        nS = self.get_number_of_raytracing_solutions()
+        dictionary['ray_tracing_C0'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
+        dictionary['ray_tracing_C1'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
+        dictionary['focusing_factor'] = np.ones((n_showers, n_antennas, nS))
+        dictionary['ray_tracing_reflection'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
+        dictionary['ray_tracing_reflection_case'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
+        dictionary['ray_tracing_solution_type'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
+
+    def get_ray_tracing_perfomed(self, station_dictionary, station_id):
```

Comment:
I guess this is what you meant with the raytracer now having a station ID? I don't see another solution here, since the ray_tracing_C0 property is specific to this raytracer, so we should keep it out of simulation.py, but we need to know the station for which we need to check for the solution

Reply:
If you move the filling of hdf5 files (or the data structure in general) to simulation.py (how it was previously) you can just check if you have an entry in your data for this station id. If you do have it, initialize the ray tracer accordingly.

Reply:
But just having an entry for the station ID doesn't necessarily mean that that the raytracing was actually performed, right? The entry for the station is created at the start and filled with nan, so we would need to check if the parameters for the raytracing solution are set. The problem is that the C_0 and C_1 parameters are specific to the analytic raytracer, so we should check for them there.
How about if simulation.py passes only the entry for the station (station_dictionary[station_id]) to the raytracer and it checks if the parameters for its solution are set? That way the raytracer doesn't need to know about the station IDs and the simulation.py doesn't need to know anything specific about the raytracer.

Reply:
maybe I'm missing something here, but wouldn't you need to first read the ray tracer solution values from the hdf5 file and initialize the ray tracer with these values? So you always need to check the hdf5 file first. 
If you have a nan in the hdf5 file, then this is a correct information, because it means that no ray tracing solution was found for this parameter.

Reply:
I don't think this is ever set in the raytracer. The all the entries in the input hdf are copied into the output data structure, but that is simply done by looping over its keys, so we don't need to hardcode the key names there. If get_raytracing_performed returns True, that means the raytracing solution is already in the output dict, so we don't need to do anything else.